neuralop/models/rno.py

```diff
@@ -0,0 +1,164 @@
+                self.domain_padding = DomainPadding(domain_padding=domain_padding, padding_mode=domain_padding_mode, output_scaling_factor=output_scaling_factor)
+        else:
+            self.domain_padding = None
+
+        self.domain_padding_mode = domain_padding_mode
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.lifting_channels = lifting_channels
+        self.projection_channels = projection_channels
+        self.residual = residual
+
+        # if lifting_channels is passed, make lifting an MLP with a hidden layer of size lifting_channels
+        if self.lifting_channels:
+            self.lifting = MLP(in_channels=in_channels, out_channels=self.hidden_channels, hidden_channels=self.lifting_channels, n_layers=2, n_dim=self.n_dims)
+        # otherwise, make it a linear layer
+        else:
+            self.lifting = MLP(in_channels=in_channels, out_channels=self.hidden_channels, hidden_channels=self.hidden_channels, n_layers=1, n_dim=self.n_dims)
+
+        module_list = [RNO_layer(n_modes, hidden_channels, return_sequences=True, output_scaling_factor=self.output_scaling_factor[i], fft_norm=fft_norm, factorization=factorization, separable=separable)
+                                     for i in range(n_layers - 1)]
+        module_list.append(RNO_layer(n_modes, hidden_channels, return_sequences=False, output_scaling_factor=self.output_scaling_factor[-1], fft_norm=fft_norm, factorization=factorization, separable=separable))
+        self.layers = nn.ModuleList(module_list)
+
+        # if projection_channels is passed, make lifting an MLP with a hidden layer of size lifting_channels
+        if self.projection_channels:
+            self.projection = MLP(in_channels=self.hidden_channels, out_channels=out_channels, hidden_channels=self.projection_channels, n_layers=2, n_dim=self.n_dims)
+        # otherwise, make it a linear layer
+        else:
+            self.projection = MLP(in_channels=self.hidden_channels, out_channels=out_channels, hidden_channels=out_channels, n_layers=1, n_dim=self.n_dims)
+    
+    def forward(self, x, init_hidden_states=None): # h must be padded if using padding
+        # x shape (batch, timesteps, dim, dom_size1, dom_size2, ..., dom_sizen)
+        batch_size, timesteps = x.shape[:2]
+        dim = x.shape[2]
+        dom_sizes = x.shape[3 : 3 + self.n_dims]
+        x_size = len(x.shape)
+
+        if init_hidden_states is None:
+            init_hidden_states = [None] * self.n_layers
+        
+        x = self.lifting(x.reshape(batch_size * timesteps, *x.shape[2:]))
+        x = x.reshape(batch_size, timesteps, *x.shape[1:])
+
+        if self.domain_padding:
+            x = self.domain_padding.pad(x)
+
+        final_hidden_states = []
+        for i in range(self.n_layers):
+            pred_x = self.layers[i](x, init_hidden_states[i])
+            if i < self.n_layers - 1:
+                if self.residual:
+                    x = x + pred_x
```

Comment:
I think you need to apply self.layers[i].transform to x in case of output_scaling factor passed by user

Reply:
I don't think I have implemented a `transform()` function for the recurrent layers. Do I need to do this? I thought any `output_scale_factor` differences were handled by the Fourier blocks under the hood.

Reply:
Yes but what happens if self.layers[i] downscales (or upscales) the input? Or does this never happen?

Reply:
If I understand correctly, we need `transform` after each spectral convolution in case we upscale/downscale the input? In `recurrent_layers.py`, I operate directly on FNO blocks, so I think any changes in resolution are dealt with inside the blocks. What do you think?